src/cmap/test/event.rs

```diff
@@ -1,29 +1,27 @@
-use std::sync::{Arc, RwLock};
+use std::{
+    collections::VecDeque,
+    sync::{Arc, RwLock},
+};
 
 use serde::{de::Unexpected, Deserialize, Deserializer};
 
 use crate::{event::cmap::*, options::StreamAddress};
 
 #[derive(Clone, Debug)]
 pub struct EventHandler {
-    ignore: Vec<String>,
-    pub events: Arc<RwLock<Vec<Event>>>,
+    pub events: Arc<RwLock<VecDeque<Event>>>,
 }
 
 impl EventHandler {
-    pub fn new(ignore: Vec<String>) -> Self {
```

Comment:
According to the CMAP spec:
>It is important to note that the ignore list is used for calculating actualEvents, but is NOT used for the waitForEvent command

This was causing us to fail some of the newer spec tests, so I fixed it to only do this at the end.

Reply:
Oh, interesting, I hadn't realized that